cli/script/command-executor.ts

```diff
@@ -509,18 +599,83 @@ function getAccessKeyId(accessKeyName: string): Promise<string> {
 }
 
 function getApp(appName: string): Promise<App> {
+    var ownerEmailValue: string;
+    var appNameValue: string = appName;
+    var delimiterIndex: number = appName.indexOf("/");
+
+    if (delimiterIndex !== -1) {
+        ownerEmailValue = appName.substring(0, delimiterIndex);
+        appNameValue = appName.substring(delimiterIndex + 1);
+
+        throwForInvalidEmail(ownerEmailValue);
+    }
+
     return sdk.getApps()
         .then((apps: App[]): App => {
+            var foundApp = false;
+            var possibleApp: App;
+
             for (var i = 0; i < apps.length; ++i) {
                 var app: App = apps[i];
+                if (app.name === appNameValue) {
+                    var isCurrentUserOwner: boolean = isCurrentAccountOwner(app.collaborators);
+                    if (ownerEmailValue) {
+                        var appOwner: string = getOwnerEmail(app.collaborators);
+                        foundApp = appOwner && appOwner === ownerEmailValue;
+                    } else if (!isCurrentUserOwner) {
+                        // found an app name matching given value but is not the owner of the app
+                        // its possible there is another app with same name of which this user 
+                        // is the owner, so keep this pointer for future use.
+                        possibleApp = app;
+                    } else {
+                        foundApp = isCurrentUserOwner;
+                    }
+                }
 
-                if (app.name === appName) {
+                if (foundApp) {
                     return app;
                 }
             }
+
+            if (possibleApp) {
+                return possibleApp;
+            }
         });
 }
 
+function isCurrentAccountOwner(map: CollaboratorMap): boolean {
+    if (map) {
+        var ownerEmail: string = getOwnerEmail(map);
+        return ownerEmail && !!map[ownerEmail].isCurrentAccount;
```

Comment:
`isCurrentAccount` could be undefined or true, my premise for adding `!!` is to make it explicit that it could be undefined. I know it doesn't matter in `if(...)` or `&&` comparators, having it explicit makes it more obvious for me. Nonetheless, I don't feel too strongly about it, I will remove it.